Implement scrollbar-width CSS property

lukewarlow opened the pull request:
#### 1cb3112c19cbc64cf2a0ff0ee8f202f23186dc60
<pre>
Implement scrollbar-width CSS property
<a href="https://bugs.webkit.org/show_bug.cgi?id=231588">https://bugs.webkit.org/show_bug.cgi?id=231588</a>

Reviewed by NOBODY (OOPS!).

This change adds an implementation of the scrollbar-width CSS property
behind an off by default flag.
All three values (auto, thin, and none) are supported.
Thin has no effect inside of WebKit legacy but this is spec compliant.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/auto-scrollbar-inline-children-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/auto-scrollbar-inline-children.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/multicol-in-orthogonal-writing-mode-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-001-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-002-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-003-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-004-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-005-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-005.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-006-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-006.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-007-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-007.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-008-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-008.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-keywords-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-keywords.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-parsing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-parsing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/viewport-scrollbar-body-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/viewport-scrollbar-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/scrollbar-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/client-props-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::createScrollbar):
(WebCore::LocalFrameView::scrollbarWidth const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::scrollbarWidth const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::createScrollbar):
(WebCore::RenderLayerScrollableArea::scrollbarWidth const):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderScrollbar.cpp:
(WebCore::RenderScrollbar::isHiddenByStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::initialScrollbarWidth):
(WebCore::RenderStyle::setScrollbarWidth):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::scrollbarWidth const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
(WebCore::StyleRareNonInheritedData::StyleRareNonInheritedData):
(WebCore::StyleRareNonInheritedData::operator== const):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
(-[WebDynamicScrollBarsView updateScrollers]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _scrollbarWidthNone]):
* Source/WebKitLegacy/mac/WebView/WebHTMLViewPrivate.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1cb3112c19cbc64cf2a0ff0ee8f202f23186dc60

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6628 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6845 "Failed to compile WebKit") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7026 "Failed to compile WebKit") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8219 "Failed to compile WebKit") | [❌ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6913 "Failed to compile WebKit") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6627 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/7910 "Failed to compile WebKit") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6792 "Failed to compile WebKit") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/8219 "Failed to compile WebKit") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6746 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/23/builds/7910 "Failed to compile WebKit") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/14/builds/7026 "Failed to compile WebKit") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8306 "Failed to compile WebKit") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/23/builds/7910 "Failed to compile WebKit") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/14/builds/7026 "Failed to compile WebKit") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/8306 "Failed to compile WebKit") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/5598 "Built successfully and passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/23/builds/7910 "Failed to compile WebKit") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/14/builds/7026 "Failed to compile WebKit") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/8306 "Failed to compile WebKit") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6216 "Built successfully and passed tests") | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6565 "Failed to compile WebKit") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/16/builds/6792 "Failed to compile WebKit") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/6765 "Built successfully") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5961 "Failed to compile WebKit") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/14/builds/7026 "Failed to compile WebKit") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1570 "Passed tests") | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10132 "Failed to compile WebKit") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/6948 "Built successfully") | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6334 "Failed to compile WebKit") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1699 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

graouts commented on Source/WebCore/animation/CSSPropertyAnimation.cpp:
```diff
@@ -3744,6 +3744,7 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
         case CSSPropertyScrollSnapAlign:
         case CSSPropertyScrollSnapStop:
         case CSSPropertyScrollSnapType:
+        case CSSPropertyScrollbarWidth:
```
This property is [animatable according to its spec](https://w3c.github.io/csswg-drafts/css-scrollbars/#scrollbar-width). You should be able to make it animatable by adding the following line of code along where other `DiscretePropertyWrapper` objects are created:

```c+
new DiscretePropertyWrapper<ScrollbarWidth>(CSSPropertyScrollbarWidth, &RenderStyle::scrollbarWidth, &RenderStyle::setScrollbarWidth)
```

Animation support would also need to be tested, as it does not seem to be the case as it stands. You could do that by adding a new type to `LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js`. Looking at that file, it should be pretty straightforward for you to add a new discrete animatable property.

Let me know if you need help.

graouts commented on LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
```diff
@@ -1172,6 +1172,12 @@ const gCSSProperties2 = {
+  'scrollbar-width': {
+    // https://w3c.github.io/csswg-drafts/css-scrollbars/#propdef-scrollbar-width
+    types: [
+      { type: 'discrete', options: [ [ 'auto', 'thin', 'none' ] ] }
```
I think you can only provide pairs of values here, so if you want to try all three, you'd have to specify the three combinations.

lukewarlow replied:
Just noticed this issue locally have changed to to just be thin and auto.

graouts replied:
I think it's possible you may have a bug with `auto` values per the bot results I saw for the WPT `web-animations` tests. You should probably still test `auto` with other values and see what gives.

graouts commented on Source/WebCore/animation/CSSPropertyAnimation.cpp:
```diff
@@ -3545,7 +3545,8 @@ CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap()
         new DiscreteSVGPropertyWrapper<const String&>(CSSPropertyMarkerMid, &SVGRenderStyle::markerMidResource, &SVGRenderStyle::setMarkerMidResource),
-        new DiscreteSVGPropertyWrapper<const String&>(CSSPropertyMarkerStart, &SVGRenderStyle::markerStartResource, &SVGRenderStyle::setMarkerStartResource)
+        new DiscreteSVGPropertyWrapper<const String&>(CSSPropertyMarkerStart, &SVGRenderStyle::markerStartResource, &SVGRenderStyle::setMarkerStartResource),
+        new DiscretePropertyWrapper<ScrollbarWidth>(CSSPropertyScrollbarWidth, &RenderStyle::scrollbarWidth, &RenderStyle::setScrollbarWidth)
```
Very nice.

lukewarlow commented on Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
```diff
@@ -870,10 +870,13 @@ Ref<Scrollbar> RenderLayerScrollableArea::createScrollbar(ScrollbarOrientation o
     bool hasCustomScrollbarStyle = is<RenderBox>(actualRenderer) && downcast<RenderBox>(actualRenderer).style().hasPseudoStyle(PseudoId::Scrollbar);
     auto element = downcast<RenderBox>(actualRenderer).element();
-    if (hasCustomScrollbarStyle && element)
+    if ((hasCustomScrollbarStyle || scrollbarWidth() == ScrollbarWidth::None) && element)
```
I think ideally this would just go down the native scrollbar code path? And then hide it inside of Scrollbar.cpp but when I tried this I couldn't get it working. It would still reserve space and show when stationary but flicker away on scroll.

Ahmad-S792 replied:
https://bugs.webkit.org/show_bug.cgi?id=249278 ?

lukewarlow replied:
That bug definitely sounds related to this issue.

lukewarlow replied:
I think in this case the issue may have been to do with the fact that the scrollbar thickness was being incorrectly set to full width and this messed with the rendering despite nativeScrollbarVisibility being set to HiddenByStyle.

Will hopefully be able to fully implement it using the native codepath when I make the new PR that implements the code (I'm splitting this one up)

graouts commented on LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
```diff
@@ -1172,6 +1172,12 @@ const gCSSProperties2 = {
       { type: 'discrete', options: [ [ 'auto', 'smooth' ] ] }
     ]
   },
+  'scrollbar-width': {
+    // https://w3c.github.io/csswg-drafts/css-scrollbars/#propdef-scrollbar-width
+    types: [
+      { type: 'discrete', options: [ [ 'auto', 'thin' ], [ 'auto', 'none' ], [ 'thin', 'none' ] ] }
+    ]
+  },
```
That's excellent.

graouts commented on LayoutTests/TestExpectations:
```diff
@@ -6271,3 +6271,16 @@ webkit.org/b/252183 imported/w3c/web-platform-tests/css/css-box/margin-trim/bloc
+imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003.html [ Skip ]
+
+#This test relies on textarea resize handle showing without a scrollbar which isn't implemented yet.
+imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none.html [ Skip ]
```
Ideally these would refer to bugzilla bugs tracking their completion.

lukewarlow replied:
Gonna have another look through this PR to try and fix the pipelines up today. Will also make or find bugs to link to for the skipped tests

lukewarlow committed:
Implement scrollbar-width CSS property
https://bugs.webkit.org/show_bug.cgi?id=231588

Reviewed by NOBODY (OOPS!).

This change adds an implementation of the scrollbar-width CSS property
behind an off by default flag.
All three values (auto, thin, and none) are supported.
Thin has no effect inside of WebKit legacy but this is spec compliant.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/META.yml: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/auto-scrollbar-inline-children-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/auto-scrollbar-inline-children.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/multicol-in-orthogonal-writing-mode-crash.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-001-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-002-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-003-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-004-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-004.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-005-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-005.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-006-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-006.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-007-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-007.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-008-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-008.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-keywords-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-keywords.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-002.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-paint-003.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-parsing-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-parsing.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbars-chrome-bug-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/viewport-scrollbar-body-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/viewport-scrollbar-frame.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/support/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/textarea-scrollbar-width-none.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/transparent-on-root.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-body.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/viewport-scrollbar.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/scrollbar-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/cssom-view/client-props-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::createScrollbar):
(WebCore::LocalFrameView::scrollbarWidth const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/platform/ScrollableArea.h:
(WebCore::ScrollableArea::scrollbarWidth const):
* Source/WebCore/rendering/RenderLayerScrollableArea.cpp:
(WebCore::RenderLayerScrollableArea::createScrollbar):
(WebCore::RenderLayerScrollableArea::scrollbarWidth const):
* Source/WebCore/rendering/RenderLayerScrollableArea.h:
* Source/WebCore/rendering/RenderScrollbar.cpp:
(WebCore::RenderScrollbar::isHiddenByStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::initialScrollbarWidth):
(WebCore::RenderStyle::setScrollbarWidth):
* Source/WebCore/rendering/style/RenderStyle.h:
(WebCore::RenderStyle::scrollbarWidth const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator<<):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
(WebCore::StyleRareNonInheritedData::StyleRareNonInheritedData):
(WebCore::StyleRareNonInheritedData::operator== const):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebKitLegacy/mac/WebView/WebDynamicScrollBarsView.mm:
(-[WebDynamicScrollBarsView updateScrollers]):
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(-[WebHTMLView _scrollbarWidthNone]):
* Source/WebKitLegacy/mac/WebView/WebHTMLViewPrivate.h:
